jobs/integration/conftest.py

```diff
@@ -150,13 +150,19 @@ async def run(self, cmd, *args, stdin=None):
             )
         return stdout.decode("utf8"), stderr.decode("utf8")
 
-    async def juju_wait(self, *args, **kwargs):
-        cmd = ["/snap/bin/juju-wait", "-e", self.connection, "-w", "-v"]
-        if args:
-            cmd.extend(args)
-        if "timeout_secs" in kwargs and kwargs["timeout_secs"]:
-            cmd.extend(["-t", str(kwargs["timeout_secs"])])
-        return await self.run(*cmd)
+    def juju_wait(self, *args, **kwargs):
```

Comment:
I'm not so sure about this. This is going to make every call to `juju_wait` block the current task without yielding to the event loop. Every task in the event loop will be blocked for potentially several minutes at a time.

I'm worried about background tasks getting blocked by this, particularly in python-libjuju which is heavily based in coroutines. See for example this [watcher task](https://github.com/charmed-kubernetes/python-libjuju/blob/fd94f0635a36518573a209a93f5361d3152c6df4/juju/model.py#L1260-L1264) that watches for model events to update model state.

Reply:
alrighty -- i'll refactor to keep using the async... but still with tee'd stdout